pkg/controllers/unsupported/unsupported_controller.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unsupported
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-capi-operator/pkg/controllers"
+	"github.com/openshift/cluster-capi-operator/pkg/operatorstatus"
+)
+
+// CapiUnsupportedController on unsupported platforms watches and keeps the cluster-api ClusterObject up to date.
+type CapiUnsupportedController struct {
```

Comment:
Unsupported, or NoOp?

Reply:
I had it NoOp and changed it to Unsupported, so it reads as CAPI Unsupported.
I think the latter is more meaningful and explicit in telling we are running in an unsupported mode.
The NoOp is quite vague and won't be true if in the future we decide to do "some op" in this controller.

Reply:
I guess my concern is whether a customer would read this and say, why is there something unsupported on my cluster. Since we use that term frequently to tell customers what they can and cannot do.

Reply:
Stick with it as it is for now, but I think having the cluster operator saying something is probably a good idea. Maybe more down the lines of "Cluster API is not implemented on this platform"

Reply:
Yes I agree with this last point, maybe something like? https://github.com/openshift/cluster-capi-operator/pull/157#discussion_r1469352816